x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/custom/CustomServiceSettings.java

```diff
@@ -162,24 +147,44 @@ public static TextEmbeddingSettings fromMap(Map<String, Object> map, TaskType ta
             SimilarityMeasure similarity = extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
             Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
             Integer maxInputTokens = removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
-            return new TextEmbeddingSettings(similarity, dims, maxInputTokens, DenseVectorFieldMapper.ElementType.FLOAT);
+            return new TextEmbeddingSettings(similarity, dims, maxInputTokens);
+        }
+
+        private final SimilarityMeasure similarityMeasure;
+        private final Integer dimensions;
+        private final Integer maxInputTokens;
+
+        public TextEmbeddingSettings(
+            @Nullable SimilarityMeasure similarityMeasure,
+            @Nullable Integer dimensions,
+            @Nullable Integer maxInputTokens
+        ) {
+            this.similarityMeasure = similarityMeasure;
+            this.dimensions = dimensions;
+            this.maxInputTokens = maxInputTokens;
         }
 
         public TextEmbeddingSettings(StreamInput in) throws IOException {
-            this(
-                in.readOptionalEnum(SimilarityMeasure.class),
-                in.readOptionalVInt(),
-                in.readOptionalVInt(),
-                in.readOptionalEnum(DenseVectorFieldMapper.ElementType.class)
-            );
+            this.similarityMeasure = in.readOptionalEnum(SimilarityMeasure.class);
+            this.dimensions = in.readOptionalVInt();
+            this.maxInputTokens = in.readOptionalVInt();
+
+            if (in.getTransportVersion().before(TransportVersions.ML_INFERENCE_CUSTOM_SERVICE_EMBEDDING_TYPE)
+                && in.getTransportVersion().isPatchFrom(TransportVersions.ML_INFERENCE_CUSTOM_SERVICE_EMBEDDING_TYPE_8_19) == false) {
+                in.readOptionalEnum(DenseVectorFieldMapper.ElementType.class);
```

Comment:
For older versions, we'll read it but ignore it. It should only be float which we'll default to in the `TextEmbeddingResponseParser`.